frontend/src/Components/Header/index.js

```diff
@@ -0,0 +1,83 @@
+import React, { useEffect, useState } from "react";
+import { Row, Col, Layout, Menu, Typography } from "antd";
+import { Link, useLocation } from "react-router-dom";
+import "./index.css";
+
+import UserStatus from "./UserStatus";
+
+const Header = () => {
+  const { Header } = Layout;
+  const { Title } = Typography;
+  const location = useLocation();
+  const [selected_keys, setSelected_keys] = useState(["0"]);
+
+  // Update menu selection based on current path
+  useEffect(() => {
+    const currentPath = location.pathname;
+    let first_dir = currentPath.substr(1).split("/")[0];
+    switch (String(first_dir)) {
+      case "":
+        setSelected_keys(["0"]);
+        break;
+      case "dashboard":
+        setSelected_keys(["1"]);
+        break;
+      case "teams":
+        setSelected_keys(["2"]);
+        break;
+      case "players":
+        setSelected_keys(["3"]);
+        break;
+      case "creator":
+        setSelected_keys(["4"]);
+        break;
+      default:
+        setSelected_keys([]);
+    }
+  }, [location]);
+
+  return (
+    <Header className="header" style={{ padding: 0 }}>
+      <Row align="center">
+        <Col span={8}>
+          <Title
+            level={5}
+            style={{
+              color: "white",
+              display: "inline-block",
+              paddingLeft: "20px",
+            }}
+          >
+            Competitive Teams
+          </Title>
+        </Col>
+        <Col span={12} align="left">
+          <Menu theme="dark" mode="horizontal" selectedKeys={selected_keys}>
+            <Menu.Item key="0">
```

Comment:
Consider putting menu elements in object/array and map it for every menu item.

e.g. 
```
const menu = [{link: '/', tabName: 'Home'}, {link: '/dashboard/profile,  tabName:'Teams'}, ....]

<Menu>
  {menu.map((menuItem, i) => (
    <Menu.Item key={i}>
      <Link to={menuItem.link}>{menuItem.tabName}</Link>
    </Menu.Item>))}
</Menu>